### PR TITLE
SPCTR-3112 Improvement: Added a hook to set font family in Spectra for Starter template plugin.

### PR DESCRIPTION
**Pull request by AbhijitNage123**
### Description

```
// Create an instance.
$fse_fonts_comp_instance = new UAGB_FSE_Fonts_Compatibility();

$body_font_family_slug = $typography_passed['body-font-family'];

$headings_font_family_slug = $typography_passed['headings-font-family'];

// Call the function using instance.
$fse_fonts_comp_instance->get_font_family_for_starter_template(array($body_font_family_slug,$headings_font_family_slug));

```

### Screenshots

- https://share.getcloudapp.com/geuPA1lL

### Types of changes

- Improvement.

### How has this been tested?

- Added to the card.

### Checklist:
- [x] My code is tested
- [ ] My code follows accessibility standards 
- [ ] My code has proper inline documentation 
- [x] I've included any necessary tests 
- [ ] I've included developer documentation 
- [x] I've added proper labels to this pull request

### CODE REVIEW

File: classes/class-uagb-fse-fonts-compatibility.php [hunk 1]
```diff
@@ -110,6 +110,85 @@ public function __construct() {
...
+				$this->add_or_update_theme_font_faces( $font_family, $font_slug, $new_font_faces );
+			}
+
+			$theme_json_raw = json_decode( file_get_contents( get_stylesheet_directory() . '/theme.json' ), true );
+
+			if ( empty( $theme_json_raw['settings']['typography']['fontFamilies'] ) ) {
+				return;
+			}
+
+			$all_font_families = $theme_json_raw['settings']['typography']['fontFamilies'];
```
Comment (adi3890) [inline classes/class-uagb-fse-fonts-compatibility.php:149]: this declaration should be done first and further check for ! empty and is_array

File: classes/class-uagb-fse-fonts-compatibility.php [hunk 2]
```diff
@@ -110,6 +110,85 @@ public function __construct() {
...
+
+			foreach ( $all_font_families as $font_families_item ) {
+				$new_font_families_item = array(
+					'value'   => $font_families_item['fontFamily'],
+					'label'   => $font_families_item['fontFamily'],
+					'weights' => array(
+						array(
+							'value' => 'Default',
+							'label' => 'Default',
```
Comment (adi3890) [inline classes/class-uagb-fse-fonts-compatibility.php:158]: should the labels be translatable

File: classes/class-uagb-fse-fonts-compatibility.php [hunk 3]
```diff
@@ -110,6 +110,85 @@ public function __construct() {
...
+					),
+				);
+
+				$new_font_families[] = $new_font_families_item;
+			}
+
+			UAGB_Admin_Helper::update_admin_settings_option( 'spectra_global_fse_fonts', $new_font_families );
+		}
+
+
```
Comment (adi3890) [inline classes/class-uagb-fse-fonts-compatibility.php:191]: Remove extra spacing

File: classes/class-uagb-fse-fonts-compatibility.php [hunk 4]
```diff
@@ -110,6 +110,85 @@ public function __construct() {
 				add_action( 'admin_init', array( $this, 'save_google_fonts_to_theme' ) );
 			}
 		}
+		/**
```
Comment (adi3890) [inline classes/class-uagb-fse-fonts-compatibility.php:113]: Please have a space after previous function

File: classes/class-uagb-fse-fonts-compatibility.php (hunk 3 above)
Comment (AbhijitNage123) [reply classes/class-uagb-fse-fonts-compatibility.php]: Done

File: classes/class-uagb-fse-fonts-compatibility.php (hunk 4 above)
Comment (AbhijitNage123) [reply classes/class-uagb-fse-fonts-compatibility.php:114]: Done

File: classes/class-uagb-fse-fonts-compatibility.php (hunk 1 above)
Comment (AbhijitNage123) [reply classes/class-uagb-fse-fonts-compatibility.php]: Done

File: classes/class-uagb-fse-fonts-compatibility.php [hunk 5]
```diff
@@ -110,6 +110,85 @@ public function __construct() {
...
+		 * Get, add and update font family in Spectra for ST.
+		 *
+		 * @param array $families font family.
+		 * @since 2.7.0
+		 */
```
[the remark this answers is not in the document]
Comment (AbhijitNage123) [reply classes/class-uagb-fse-fonts-compatibility.php:120]: Done

File: classes/class-uagb-fse-fonts-compatibility.php [hunk 6]
```diff
@@ -110,6 +110,88 @@ public function __construct() {
...
+					);
+					$this->add_or_update_theme_font_faces( $font_family, $font_slug, $new_font_faces );
+				}
+			}
+
+			$theme_json_raw = json_decode( file_get_contents( get_stylesheet_directory() . '/theme.json' ), true );
+
+			$all_font_families = $theme_json_raw['settings']['typography']['fontFamilies'];
+			
+			if ( empty( $all_font_families ) && ! is_array( $all_font_families ) ) {
```
[the remark this answers is not in the document]
Comment (AbhijitNage123) [reply classes/class-uagb-fse-fonts-compatibility.php]: Done

**Review by adi3890 — APPROVED** [review]

**Pull request merged**